universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -39,7 +43,13 @@ public ApplicantProgramBlocksController(
               Optional<Block> block = roApplicantProgramService.getBlock(blockId);
 
               if (block.isPresent()) {
-                return ok(editView.render(request, applicantId, programId, block.get()));
+                return ok(
+                    editView.render(
+                        request,
+                        messagesApi.preferred(request),
+                        applicantId,
+                        programId,
+                        block.get()));
```

Comment:
Suggestion (non-blocking): at five params I wonder if the view class should have an inner `public static Params` class that's an autovalue holding all of these. Lengthy param lists are unpleasant and having two `long` param types next to each other makes me nervous since it'd be easy to pass them in swapped.

Reply:
Interesting! I had never thought to have a parameters class, I'll look into it

Reply:
I added a Params class that has only a builder (to avoid the ambiguous `create` variable ordering). Errorprone complained that `Block` is final and doesn't override `toString` so I added a very dumb toString method - suggestions for a better one?